ee/localserver/zta_auth_middleware.go

```diff
@@ -0,0 +1,285 @@
+package localserver
+
+import (
+	"crypto/ecdsa"
+	"encoding/base64"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"log/slog"
+	"net/http"
+	"net/url"
+	"time"
+
+	"github.com/kolide/krypto/pkg/echelper"
+	"github.com/lestrrat-go/jwx/jwk"
+)
+
+type ztaAuthMiddleware struct {
+	counterPartyKeys map[string]*ecdsa.PublicKey
```

Comment:
Small documentation suggestion, since I see we document the format of `counterPartyKeys` later in chain.validate too --

```suggestion
	counterPartyKeys map[string]*ecdsa.PublicKey // maps KID to pubkey
```